aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -333,6 +315,52 @@ Tensor& index_out(Tensor& result, const Tensor & self, const torch::List<c10::op
   return result;
 }
 
+Tensor & put_(Tensor & self, const Tensor& index, const Tensor & source, const bool accumulate) {
+  // See note [Writing Nondeterministic Operations]
+  // Nondeterministic when index contains duplicate entries and we do not accumulate
```

Comment:
cc @kurtamohler 

I know previously we discussed indexing operations that may be nondeterministic based on their values. @ngimel and I were looking at this and it seems OK to simply consider them "always nondeterministic," at least for the 1.9 release. If people complain about wanting to use put_ in a deterministic context then we could consider adding a check that the values will cause the operation to be deterministic, or just implementing a deterministic put_ operation.

Follow-up: @ngimel and I both still agree that creating an overlapped tensor makes a program in error.

Reply:
That seems reasonable. I think it would be nice to aim toward eventually allowing all operations to work (when called properly, without duplicate indices) when the deterministic flag is turned on. But I also like the idea of first being conservative about what we allow when the deterministic flag is turned on, so I'm on board with considering this always nondeterministic